packaging/_parser.py

```diff
@@ -0,0 +1,155 @@
+from typing import Any, List, Tuple
+
+from ._tokenizer import Tokenizer
+
+
+def parse_named_requirement(requirement: str) -> Tuple[str, str, List[str], str, str]:
```

Comment:
It would be nice to return a namedtuple from here, so that this definition is self-contained.